lib/stream.js

```diff
@@ -93,11 +97,21 @@ const getStreamPromise = (stream, {encoding, buffer, maxBuffer}) => {
 		return;
 	}
 
-	if (encoding) {
-		return getStream(stream, {encoding, maxBuffer});
+	// eslint-disable-next-line unicorn/text-encoding-identifier-case
+	if (encoding === 'utf8' || encoding === 'utf-8') {
+		return getStream(stream, {maxBuffer});
+	}
+
+	if (encoding === null) {
+		return getStreamAsBuffer(stream, {maxBuffer});
 	}
 
-	return getStream.buffer(stream, {maxBuffer});
+	return applyEncoding(stream, maxBuffer, encoding);
+};
+
+const applyEncoding = async (stream, maxBuffer, encoding) => {
+	const buffer = await getStreamAsBuffer(stream, {maxBuffer});
+	return buffer.toString(encoding);
```

Comment:
There is one subtle change of behavior here (that is quite of an edge case): when using both `maxBuffer` and an unusual `encoding` (that is neither `utf8` nor `null`, such as `hex` or `base64`), then the number of bytes being limited by `maxBuffer` is stdout/stderr as a `Buffer`, not as the final value of `childProcess.stdout/stderr`, which is an hex/base64/etc.-encoded string.